src/lib/homestore_backend/hs_pg_manager.cpp

```diff
@@ -195,22 +197,19 @@ void HSHomeObject::on_pg_meta_blk_found(sisl::byte_view const& buf, void* meta_c
     homestore::superblk< pg_info_superblk > pg_sb(_pg_meta_name);
     pg_sb.load(buf, meta_cookie);
 
-    auto v = hs_repl_service().get_repl_dev(pg_sb->replica_set_uuid);
-    if (v.hasError()) {
-        // TODO: We need to raise an alert here, since without pg repl_dev all operations on that pg will fail
-        LOGE("open_repl_dev for group_id={} has failed", boost::uuids::to_string(pg_sb->replica_set_uuid));
-        return;
-    }
     auto pg_id = pg_sb->id;
     auto uuid_str = boost::uuids::to_string(pg_sb->index_table_uuid);
-    auto hs_pg = std::make_unique< HS_PG >(std::move(pg_sb), std::move(v.value()));
+    auto hs_pg = std::make_unique< HS_PG >(std::move(pg_sb));
     // During PG recovery check if index is already recoverd else
     // add entry in map, so that index recovery can update the PG.
     std::scoped_lock lg(index_lock_);
-    auto it = index_table_pg_map_.find(uuid_str);
-    RELEASE_ASSERT(it != index_table_pg_map_.end(), "IndexTable should be recovered before PG");
-    hs_pg->index_table_ = it->second.index_table;
-    it->second.pg_id = pg_id;
+
+    if (auto it = index_table_pg_map_.find(uuid_str); it != index_table_pg_map_.end()) {
+        hs_pg->index_table_ = it->second.index_table;
```

Comment:
this is  called when meta service start , which is the first of all services. so this will be prior to index_service start.  so I think we will never reach here.